components/raftstore/src/store/fsm/peer.rs

```diff
@@ -1027,6 +1027,13 @@ where
             self.fsm.peer.mut_store().flush_cache_metrics();
             return;
         }
+
+        if self.fsm.peer.is_leader() && self.fsm.peer.peers_start_pending_time.len() > 0 {
+            let (_, pending_after) = self.fsm.peer.peers_start_pending_time[0];
```

Comment:
`self.fsm.peer.peers_start_pending_time[0]` may not be the longest pending time, because [`swap_remove`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.swap_remove) will not preserve ordering: 
https://github.com/tikv/tikv/blob/9029a426169d06ecf131f2624caed54784f5501f/components/raftstore/src/store/peer.rs#L1280
Perhaps we should replace `swap_remove` with [`retain`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.retain)

Reply:
Nice catch~
How about using an iterator to find the longest pending time? use `retain` is too hack?